pkg/client/package_buildpack.go

```diff
@@ -90,47 +125,88 @@ func (c *Client) PackageBuildpack(ctx context.Context, opts PackageBuildpackOpti
 
 	bpURI := opts.Config.Buildpack.URI
 	if bpURI == "" {
-		return errors.New("buildpack URI must be provided")
+		return digest, errors.New("buildpack URI must be provided")
+	}
+
+	if ok, platformRootFolder := buildpack.PlatformRootFolder(bpURI, target); ok {
+		bpURI = platformRootFolder
 	}
 
 	mainBlob, err := c.downloadBuildpackFromURI(ctx, bpURI, opts.RelativeBaseDir)
 	if err != nil {
-		return err
+		return digest, err
 	}
 
 	bp, err := buildpack.FromBuildpackRootBlob(mainBlob, writerFactory, c.logger)
 	if err != nil {
-		return errors.Wrapf(err, "creating buildpack from %s", style.Symbol(bpURI))
+		return digest, errors.Wrapf(err, "creating buildpack from %s", style.Symbol(bpURI))
 	}
 
 	packageBuilder.SetBuildpack(bp)
 
+	platform := target.ValuesAsPlatform()
+
 	for _, dep := range opts.Config.Dependencies {
+		if multiArch {
+			locatorType, err := buildpack.GetLocatorType(dep.URI, opts.RelativeBaseDir, []dist.ModuleInfo{})
+			if err != nil {
+				return digest, err
+			}
+			if locatorType == buildpack.URILocator {
+				// When building a composite multi-platform buildpack all the dependencies must be pushed to a registry
+				return digest, errors.New(fmt.Sprintf("%s is not allowed when creating a composite multi-platform buildpack, push your dependencies to a registry and use 'docker://' instead", style.Symbol(dep.URI)))
```

Comment:
Does the same restriction exist for builders? How do we find the multi-arch buildpacks if they don't exist in the registry?

Reply:
Yeah, the same restrictions apply because the same logic is executed. I think I need to add some acceptance tests for builder creation